src/app/static/src/app/store/surveyAndProgram/actions.ts

```diff
@@ -62,6 +66,41 @@ export const actions: ActionTree<SurveyAndProgramDataState, RootState> & SurveyA
             });
     },
 
+    async deleteSurvey({commit}) {
+        await api(commit)
+            .delete("/disease/survey/")
+            .then(() => {
+                commit({type: "SurveyUpdated", payload: null});
+                commit({type: "filteredData/Reset", payload: null}, {root: true});
```

Comment:
Should this be resetting the other module steps too (modelRunResponse etc)? Or will this be done in a separate ticket? Feels like we should maybe have a utility resetStateFrom method which will reset all models subsequent to a given module (and deal with the fact that some modules aren't explicitly associated with a step e.g. filteredData)

Reply:
Yes this becomes obsolete in the next ticket https://vimc.myjetbrains.com/youtrack/issue/mrc-586

Reply:
Ha, yup, I completely forgot what the whole point of that conversation was...